subsys/event_manager/event_manager_priv.h

```diff
@@ -0,0 +1,256 @@
+
+
+#ifdef CONFIG_EVENT_MANAGER_PROFILE_EVENT_DATA
+#define _ARG_LABELS_DEFINE(...) \
+	{MEM_ADDRESS_LABEL __VA_ARGS__}
+#define _ARG_TYPES_DEFINE(...) \
+	 {MEM_ADDRESS_TYPE __VA_ARGS__}
+
+#else
+#define _ARG_LABELS_DEFINE(...) \
+	{MEM_ADDRESS_LABEL}
+#define _ARG_TYPES_DEFINE(...) \
+	 {MEM_ADDRESS_TYPE}
+
+#endif /* CONFIG_EVENT_MANAGER_PROFILE_EVENT_DATA */
+
+
+/* Declarations and definitions - for more details refer to public API. */
+#define _EVENT_INFO_DEFINE(ename, types, labels, profile_func)							\
+	const static char *_CONCAT(ename, _log_arg_labels[]) __used = _ARG_LABELS_DEFINE(labels);		\
+	const static enum profiler_arg _CONCAT(ename, _log_arg_types[]) __used = _ARG_TYPES_DEFINE(types);	\
+	const static struct event_info _CONCAT(ename, _info) __used _EM_FORCED_ALIGNMENT			\
+	__attribute__((__section__("event_infos"))) = {								\
+				.profile_fn	= profile_func,							\
+				.log_arg_cnt	= ARRAY_SIZE(_CONCAT(ename, _log_arg_labels)),			\
+				.log_arg_labels	= _CONCAT(ename, _log_arg_labels),				\
+				.log_arg_types	= _CONCAT(ename, _log_arg_types)				\
+			}
+
+
+
+#define _EVENT_LISTENER(lname, notification_fn)								\
+	const struct event_listener _CONCAT(__event_listener_, lname) __used _EM_FORCED_ALIGNMENT	\
+	__attribute__((__section__("event_listeners"))) = {						\
+		.name = STRINGIFY(lname),								\
+		.notification = (notification_fn),							\
+	}
+
+
+#define _EVENT_TYPE_DECLARE_COMMON(ename)				\
+	extern const struct event_type _CONCAT(__event_type_, ename);	\
+	_EVENT_SUBSCRIBERS_DECLARE(ename);				\
+	_EVENT_CASTER_FN(ename);					\
+	_EVENT_TYPECHECK_FN(ename)
+
+
+#define _EVENT_TYPE_DECLARE(ename)					\
+	_EVENT_TYPE_DECLARE_COMMON(ename);				\
+	_EVENT_ALLOCATOR_FN(ename)
+
+
+#define _EVENT_TYPE_DYNDATA_DECLARE(ename)				\
+	_EVENT_TYPE_DECLARE_COMMON(ename);				\
+	_EVENT_ALLOCATOR_DYNDATA_FN(ename)
+
+
+#define _EVENT_TYPE_DEFINE(ename, init_log_en, log_fn, ev_info_struct)							\
+	_EVENT_SUBSCRIBERS_DEFINE(ename);										\
+	const struct event_type _CONCAT(__event_type_, ename) __used _EM_FORCED_ALIGNMENT				\
+	__attribute__((__section__("event_types"))) = {									\
+		.name				= STRINGIFY(ename),							\
+		.subs_start	= {											\
+			[_SUBS_PRIO_FIRST]	= _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FIRST)),	\
+			[_SUBS_PRIO_NORMAL]	= _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL)),	\
+			[_SUBS_PRIO_FINAL]	= _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FINAL)),	\
+		},													\
+		.subs_stop	= {											\
+			[_SUBS_PRIO_FIRST]	= _EVENT_SUBSCRIBERS_STOP(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FIRST)),	\
+			[_SUBS_PRIO_NORMAL]	= _EVENT_SUBSCRIBERS_STOP(ename, _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL)),	\
+			[_SUBS_PRIO_FINAL]	= _EVENT_SUBSCRIBERS_STOP(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FINAL)),	\
+		},													\
+		.init_log_enable		= init_log_en,								\
+		.log_event			= log_fn,								\
+		.ev_info			= ev_info_struct,							\
```

Comment:
it should be assigned only if profiler is enabled. Now event infos are created even when not used. That should also allow to clean up double defines for some macros, eg. `MEM_ADDRESS_LABEL`